src/lib.rs

```diff
@@ -80,7 +87,7 @@ impl RelayConfig {
     }
 
     fn new(default_gateway: GatewayUri, into_client: impl Into<HttpClient>) -> Self {
-        RelayConfig { default_gateway, client: into_client.into() }
+        RelayConfig { default_gateway, client: into_client.into(), prober: Prober::default() }
```

Comment:
In the future, might this be the place to pass the client to a Prober constructor to be re-used?

Reply:
If this is not explicitly bad as-is, I'm inclined to merge and then follow up with addressing those. Just a comment! Not a requested change!